miio/huizuo.py

```diff
@@ -147,7 +312,226 @@ def set_brightness(self, level):
+        else:
+            raise HuizuoException("Your device doesn't support a fan management")
+
+    @command(
+        default_output=format_output("Fan powering off"),
+    )
+    def fan_off(self):
+        """Power fan off (only for models with fan)."""
+        if self.model in MODELS_WITH_FAN_WY or self.model in MODELS_WITH_FAN_WY2:
+            return self.set_property("fan_power", False)
+        else:
+            raise HuizuoException("Your device doesn't support a fan management")
+
+    @command(
+        click.argument("fan_level", type=int),
+        default_output=format_output("Setting fan speed level to {fan_level}"),
+    )
+    def set_fan_level(self, fan_level):
+        """Set fan speed level (only for models with fan)"""
+        if self.model in MODELS_WITH_FAN_WY or self.model in MODELS_WITH_FAN_WY2:
+            if fan_level < 0 or fan_level > 100:
+                raise HuizuoException("Invalid fan speed level: %s" % fan_level)
+
+            return self.set_property("fan_level", fan_level)
+        else:
+            raise HuizuoException("Your device doesn't support a fan management")
+
+    @command(
+        default_output=format_output("Setting fan mode to 'Basic'"),
+    )
+    def set_basic_fan_mode(self):
+        """Set fan mode to 'Basic' (only for models with fan)"""
+        if self.model in MODELS_WITH_FAN_WY or self.model in MODELS_WITH_FAN_WY2:
+            return self.set_property("fan_mode", 0)
+        else:
+            raise HuizuoException("Your device doesn't support a fan management")
+
+    @command(
+        default_output=format_output("Setting fan mode to 'Natural wind'"),
+    )
+    def set_natural_fan_mode(self):
+        """Set fan mode to 'Natural wind' (only for models with fan)"""
+        if self.model in MODELS_WITH_FAN_WY or self.model in MODELS_WITH_FAN_WY2:
+            return self.set_property("fan_mode", 1)
+        else:
+            raise HuizuoException("Your device doesn't support a fan management")
+
+    # Fan Reverse option is not available for all models with fan
+    @command(
+        default_output=format_output("Enable fan reverse"),
+    )
+    def fan_reverse_on(self):
+        """Enable fan reverse (only for models which support this fan option)"""
+        if self.model in MODELS_WITH_FAN_WY:
+            return self.set_property("fan_motor_reverse", True)
+        else:
+            raise HuizuoException("Your device doesn't support a fan management")
+
+    @command(
+        default_output=format_output("Disable fan reverse"),
+    )
+    def fan_reverse_off(self):
+        """Disable fan reverse (only for models which support this fan option)"""
+        if self.model in MODELS_WITH_FAN_WY:
+            return self.set_property("fan_motor_reverse", False)
+        else:
+            raise HuizuoException("Your device doesn't support a fan management")
+
+    # The next section contains the heater management commands
```

Comment:
The heater commands should be implemented in the new device class for heaters f.e.

Reply:
Split with classes